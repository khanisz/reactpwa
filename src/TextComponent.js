import React from "react";
import PropTypes from "prop-types";
import { withStyles } from "@material-ui/core/styles";
import Card from "@material-ui/core/Card";
import CardActions from "@material-ui/core/CardActions";
import CardContent from "@material-ui/core/CardContent";
import Button from "@material-ui/core/Button";
import Typography from "@material-ui/core/Typography";

const styles = {
  card: {
    minWidth: 275,
    margin: 10
  },
  bullet: {
    display: "inline-block",
    margin: "0 2px",
    transform: "scale(0.8)"
  },
  title: {
    fontSize: 14
  },
  pos: {
    marginBottom: 12
  }
};

function TextComponent(props) {
  const { classes } = props;
  const bull = <span className={classes.bullet}>•</span>;
  const { text } = props;

  return (
    <Card className={classes.card}>
      <CardContent>
        <Typography
          className={classes.title}
          color="textSecondary"
          gutterBottom
        />
        <Typography variant="h5" component="h2">
          {text}
        </Typography>
        <Typography className={classes.pos} color="textSecondary" />
        <Typography component="p">
          Breslau, 1938 rok. Przenieście się w czasie i poznajcie największą
          tajemnicę przedwojennego Wrocławia. Znika kobieta podejrzana o zdradę
          oraz śledzący ją na zlecenie męża detektywi. Teraz wszystko w Waszych
          rękach. Zanim jednak odnajdziecie trop, najpierw musicie odnaleźć
          siebie nawzajem w podziemiach Starego Miasta...
        </Typography>
      </CardContent>
    </Card>
  );
}

TextComponent.propTypes = {
  classes: PropTypes.object.isRequired
};

export default withStyles(styles)(TextComponent);
