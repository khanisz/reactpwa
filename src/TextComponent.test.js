import React from "react";
import { shallow } from "enzyme";
import TextComponent from "./TextComponent";

describe("TextComponent", () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<TextComponent debug />);

    expect(component).toMatchSnapshot();
  });

  it("should render correctly with no props", () => {
    const component = shallow(<TextComponent />);

    expect(component).toMatchSnapshot();
  });

  it("should render correctly with title", () => {
    const title = "Tytul";
    const component = shallow(<TextComponent title={title} />);
    expect(component).toMatchSnapshot();
  });

  it("should render correctly with text", () => {
    const text = "Lore ipsum";
    const component = shallow(<TextComponent text={text} />);
    expect(component).toMatchSnapshot();
  });

  it("should render correctly with title and text", () => {
    const title = "Tytul";
    const text = "Lore ipsum";
    const component = shallow(<TextComponent title={title} text={text} />);
    expect(component).toMatchSnapshot();
  });
});
