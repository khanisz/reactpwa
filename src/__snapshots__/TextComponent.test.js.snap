// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextComponent should render correctly in "debug" mode 1`] = `
<TextComponent
  classes={
    Object {
      "bullet": "TextComponent-bullet-2",
      "card": "TextComponent-card-1",
      "pos": "TextComponent-pos-4",
      "title": "TextComponent-title-3",
    }
  }
  debug={true}
/>
`;

exports[`TextComponent should render correctly with no props 1`] = `
<TextComponent
  classes={
    Object {
      "bullet": "TextComponent-bullet-2",
      "card": "TextComponent-card-1",
      "pos": "TextComponent-pos-4",
      "title": "TextComponent-title-3",
    }
  }
/>
`;

exports[`TextComponent should render correctly with text 1`] = `
<TextComponent
  classes={
    Object {
      "bullet": "TextComponent-bullet-2",
      "card": "TextComponent-card-1",
      "pos": "TextComponent-pos-4",
      "title": "TextComponent-title-3",
    }
  }
  text="Lore ipsum"
/>
`;

exports[`TextComponent should render correctly with title 1`] = `
<TextComponent
  classes={
    Object {
      "bullet": "TextComponent-bullet-2",
      "card": "TextComponent-card-1",
      "pos": "TextComponent-pos-4",
      "title": "TextComponent-title-3",
    }
  }
  title="Tytul"
/>
`;

exports[`TextComponent should render correctly with title and text 1`] = `
<TextComponent
  classes={
    Object {
      "bullet": "TextComponent-bullet-2",
      "card": "TextComponent-card-1",
      "pos": "TextComponent-pos-4",
      "title": "TextComponent-title-3",
    }
  }
  text="Lore ipsum"
  title="Tytul"
/>
`;
